DotnetFoundation/DotnetFoundation.Api/Controllers/TaskController.cs

```diff
@@ -20,104 +20,180 @@ public TaskController(ITaskDetailsService TaskDetailsService)
-
-        response.Data = await _taskDetailsService.GetTaskByIdAsync(taskId).ConfigureAwait(false);
-        response.Status = ResponseStatus.Success;
-
-        return Ok(response);
+        try
+        {
+            response.Data = await _taskDetailsService.GetTaskByIdAsync(taskId).ConfigureAwait(false);
+            response.Status = ResponseStatus.Success;
+
+            return Ok(response);
+        }
+        catch (NotFoundException ex)
+        {
+            response.Message = ex.Message;
+            response.Status = ResponseStatus.Error;
+            return BadRequest(response);
+        }
+        catch (Exception ex)
+        {
+            response.Message = ex.Message;
+            response.Status = ResponseStatus.Error;
+            return StatusCode(StatusCodes.Status500InternalServerError, response);
+        }
     }
 
     /// <summary>
     /// Add new task.
     /// </summary>
     /// <param name="detailRequest">Role request details</param>
-    [HttpPost]    
+    [HttpPost("task")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<ActionResult<BaseResponse<TaskDetailsResponse>>> InsertTaskAsync(TaskDetailsRequest detailRequest)
     {
         BaseResponse<TaskDetailsResponse> response = new(ResponseStatus.Fail);
-
-        response.Data = await _taskDetailsService.InsertTaskAsync(detailRequest).ConfigureAwait(false);
-        response.Status = ResponseStatus.Success;
-
-        return Ok(response);
+        try
+        {
+            response.Data = await _taskDetailsService.InsertTaskAsync(detailRequest).ConfigureAwait(false);
+            response.Status = ResponseStatus.Success;
+
+            return Ok(response);
+        }
+        catch (NotFoundException ex)
+        {
+            response.Message = ex.Message;
+            response.Status = ResponseStatus.Error;
+            return BadRequest(response);
+        }
+        catch (Exception ex)
+        {
+            response.Message = ex.Message;
+            response.Status = ResponseStatus.Error;
+            return StatusCode(StatusCodes.Status500InternalServerError, response);
+        }
     }
 
     /// <summary>
     /// Update details of a task when the Id is passed.
     /// </summary>
     /// <param name="taskId">Id of task record</param>
     /// <param name="modifiedDetails">Modified details for task record</param>
-    [HttpPut("{taskId}")]    
+    [HttpPut("tasks/{taskId}")]
     [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<ActionResult<BaseResponse<TaskDetailsResponse>>> UpdateTaskAsync(int taskId, TaskDetailsRequest modifiedDetails)
```

Comment:
`modifiedDetails` is not a meaningful variable name. The variable has to be related to task.

Reply:
Renamed this to `updatedTaskDetails`